### PR TITLE
{AzureContainerApp} fixes Azure/azure-cli-extensions#5743 PermissionError for temp file

### PR DESCRIPTION
**Pull request by navba-MSFT**
fixes Azure/azure-cli-extensions#5743 PermissionError for temp file

We get the below error When executing az containerapp up without Dockerfile on Windows OS, PermissionError occurs at the ACR task creation step. It seems that temporary files cannot be accessed. This issue can be reproduced on Windows Sandbox also. And it was succeeded to deploy this app from WSL(Ubuntu 22.04) by executing the same command.

```
  File "C:\Users\WDAGUtilityAccount\.azure\cliextensions\containerapp\azext_containerapp\_up_utils.py", line 407, in run_acr_build
    self.build_container_from_source(image_name, source)
  File "C:\Users\WDAGUtilityAccount\.azure\cliextensions\containerapp\azext_containerapp\_up_utils.py", line 375, in build_container_from_source
    acr_task_create(self.cmd, task_client, task_name, registry_name, context_path="/dev/null", file=task_file.name)
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/acr/task.py", line 137, in acr_task_create
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/acr/task.py", line 292, in create_task_step
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/acr/_utils.py", line 266, in get_yaml_template
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\WDAGUT~1\\AppData\\Local\\Temp\\tmpaa2d2k9z'
```

---

Since the behavior of NamedTemporaryFile depends on OS. Hence this is failing on windows and works on WSL Ubuntu.
This PR fixes this issue.

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AzureContainerApp

**Comment by horihiro** [comment]
It seems that there is no code for removing the temp file after creating an ACR task. Isn't it needed?

**Comment by navba-MSFT** [comment]
> It seems that there is no code for removing the temp file after creating an ACR task. Isn't it needed?

@horihiro Thanks for the suggestion. I have committed the suggestion now. It is currently under review.

**Comment by horihiro** [comment]
@navba-MSFT 
I think removing the temp file by `os.unlink` in `finally` block needs to be located **after** calling `acr_task_create`.
\# And `import os` is needed to call `os.unlink`.

File: src/containerapp/setup.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
 
-VERSION = '0.3.20'
+VERSION = '0.3.21'
```
Comment (Juliehzl) [inline src/containerapp/setup.py:20]: could you hold on the version upgrade here but keeping release note in history? In this way, we could release multiple features together.

Comment (navba-MSFT) [reply src/containerapp/setup.py]: Done. Reverted the version change. Retained the history notes.

**Comment by navba-MSFT** [comment]
@Juliehzl Thanks for reviewing this. Your suggestion has been incorporated and the tests are passed.

**Review by Juliehzl — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**